README.md

```diff
@@ -61,9 +61,47 @@ Packages that are not bound to an organization or a product can use the naming c
 
 ## NuGet package
 
-NuGet package should follow [Microsoft guidelines](https://docs.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices)
+### Guidelines
 
+NuGet package should follow [Microsoft guidelines](https://docs.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices)
 
+### Folder Structure
+
+Here is the proposed folder structure that should be used:
+``` 
+.
+|-build (optional)
+|      |-build.cmd (optional)
+|      |-build.ps1 (optional)
+|      |-...(other build artefact)
+|-src
+|   |-project1
+|   |-project2
+|   |...
+|   |-projectN
+|-docs
+|    |-images
+|    |-chapter1
+|    |-chapter2
+|    |-...
+|    |-chapterN
+|-tests
+|     |-testproject1
+|     |-testproject2
+|     |-..
+|     |-testprojectN
+|-samples
+|       |-sampleprojevt1
+|       |-sampleproject2
+|       |-..
+|       |-sampleprojectN
+|-README.md (Introduction document)
+|-CHANGELOG.md (All release notes documented)
+|-CONTRIBUTING.md (Explanation on how to contribute)
+|-CODE_OF_CONDUCT.md (Rule to repect for contribution)
+|-LICENSE.txt (Apache 2.0 licence text)
+|-<solution>.sln
+```
```

Comment:
I am not sure about this. I feel like it's goes too far. There is a lot of reasons why a project could want to adopt a different structure and I don't see the benefit of standardizing this.

What's mandatory is having a LICENSE at the root. Otherwise, IMO it's up to the maintainers.

Reply:
It's a **proposed** folder structure based on what is pretty common in the market.

Reply:
Well if it's common in the market, the maintainers should tend to this structure by themselve. I don't see the value of standardizing a folder structure, there are so many things which could justify using another folder structure.

Reply:
Maybe instead of proposing a file structure we could emphasis on a few important key points that are included in your proposition like the README, CONTRIBUTING, LICENSE & CODE_OF_CONDUCT files?

For LICENSE we already have a template but we could offer one for CODE_OF_CONDUCT.

What do you think?